plot useLayoutSize() crashes on infinite recursion

### Explain what happens

1. Click the "Storage" or "Networking" tab
2. Load to a blank screen with the "Oops!" error hint at the top right

This issue does not manifest in Firefox. Ungoogled Chromium (flatpak 122.0.6261.111)

#### Storage error:
```
storaged.js:8 Uncaught Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at oi (storaged.js:8:33845)
    at OQ (storaged.js:6:24880)
    at storaged.js:655:511
    at $h (storaged.js:8:24270)
    at tB (storaged.js:8:31626)
    at YT (storaged.js:8:31478)
    at VQ (storaged.js:8:31016)
    at YQ (storaged.js:8:40923)
    at Rc (storaged.js:8:40281)
    at aB (storaged.js:8:36889)
```

#### Networking error
```
networkmanager.js:8 Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at bn (networkmanager.js:8:33829)
    at hD (networkmanager.js:6:24854)
    at networkmanager.js:17:509
    at kv (networkmanager.js:8:24259)
    at pI (networkmanager.js:8:31611)
    at o2 (networkmanager.js:8:31463)
    at OD (networkmanager.js:8:31001)
    at AD (networkmanager.js:8:40897)
    at hi (networkmanager.js:8:40257)
    at dI (networkmanager.js:8:36867)
```

### Version of Cockpit

311

### Where is the problem in Cockpit?

Storage, Networking

### Server operating system

Fedora

### Server operating system version

39

### What browsers are you using?

Other

### System log

_No response_

--- a/pkg/lib/cockpit-components-plot.jsx
+++ b/pkg/lib/cockpit-components-plot.jsx
@@ -295,7 +295,11 @@ const useLayoutSize = (init_width, init_height) => {
     useLayoutEffect(() => {
         if (ref.current) {
             const rect = ref.current.getBoundingClientRect();
-            if (rect.width != size.width || rect.height != size.height)
+            // Some browsers, such as Bromite, add noise to the result
+            // of getBoundingClientRect in order to deter
+            // fingerprinting. Let's allow for that by only reacting
+            // to significant changes.
+            if (Math.abs(rect.width - size.width) > 2 || Math.abs(rect.height - size.height) > 2)
                 setSize({ width: rect.width, height: rect.height });
         }
     });

